pootle/apps/pootle_misc/checks.py

```diff
@@ -266,56 +267,82 @@ class SkipCheck(Exception):
     pass
 
 
-class ENChecker(checks.TranslationChecker):
+class ENChecker(checks.UnitChecker):
+
+    def __init__(self, checkerconfig=None, excludefilters=None,
+                 limitfilters=None, errorhandler=None):
+        super(ENChecker, self).__init__(checkerconfig, excludefilters,
+                                        limitfilters, errorhandler)
+
+    def run_test(self, test, unit):
+        """Runs the given test on the given unit.
+
+        Overrides `TranslationChecker`'s method to avoid extra plural handling
+        (there are no gettext-style plural units in Evernote PO files), and
+        makes language information available to checks that want to make use of
+        it.
+        """
+        return test(self.str1, self.str2, language_code=self.language_code)
```

Comment:
Yes, it is able to do so, however such implicit contract/internal state makes it harder to test checks independently, as opposed to passing an explicit argument.